server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -49,25 +56,29 @@ runHasuraGQ
   :: ( MonadIO m
      , MonadError QErr m
      , MonadReader E.ExecutionCtx m
+     , MonadState QueryLog m
      )
-  => RequestId
-  -> GQLReqUnparsed
-  -> UserInfo
+  => UserInfo
   -> E.ExecOp
   -> m EncJSON
-runHasuraGQ reqId query userInfo resolvedOp = do
-  E.ExecutionCtx logger _ pgExecCtx _ _ _ _ _ <- ask
-  respE <- liftIO $ runExceptT $ case resolvedOp of
+runHasuraGQ userInfo resolvedOp = do
+  E.ExecutionCtx{_ecxPgExecCtx} <- ask
+  respE <- case resolvedOp of
     E.ExOpQuery tx genSql  -> do
-      -- log the generated SQL and the graphql query
-      liftIO $ logGraphqlQuery logger $ QueryLog query genSql reqId
-      runLazyTx' pgExecCtx tx
+      addToLog genSql
+      liftIO $ runExceptT $ runLazyTx' _ecxPgExecCtx tx
     E.ExOpMutation tx -> do
-      -- log the graphql query
-      liftIO $ logGraphqlQuery logger $ QueryLog query Nothing reqId
-      runLazyTx pgExecCtx $ withUserInfo userInfo tx
+      -- TODO: What to log here?
+      liftIO $ runExceptT $ runLazyTx _ecxPgExecCtx $ withUserInfo userInfo tx
     E.ExOpSubs _ ->
       throw400 UnexpectedPayload
       "subscriptions are not supported over HTTP, use websockets instead"
   resp <- liftEither respE
   return $ encodeGQResp $ GQSuccess $ encJToLBS resp
+  where
+    addToLog genSql =
+      modify
+        (\QueryLog {..} ->
+           let _qlGeneratedSqlMap =
```

Comment:
is this not a bug? `let x = x in x -- bottom `

Reply:
aha! I was wondering why all the queries were stuck (interesting bug though!). I think I overstretched my boundaries here with recordwildcards. I will fix this.